tests/engine/processes/test_builder.py

```diff
@@ -54,3 +58,62 @@ def test_access_methods():
 
     builder.x = node_numb
     assert dict(builder) == {'metadata': {'options': {'stash': {}}}, 'x': node_numb}
+
+
+def test_update_inputs():
+    """Test the ``ProcessBuilder._update()`` method."""
+    builder = ProcessBuilder(ArithmeticAddCalculation)
+    builder._update({'x': orm.Int(1), 'y': orm.Int(2)})
+    assert builder._inputs()['x'].value == 1
+    assert builder._inputs()['y'].value == 2
+    assert 'metadata' in builder._inputs()
+    assert 'metadata' not in builder._inputs(prune=True)
+
+    builder._update(**{'x': orm.Int(3), 'y': orm.Int(4)})
+    assert builder._inputs()['x'].value == 3
+    assert builder._inputs()['y'].value == 4
+
+    builder._update({'metadata': {'options': {'resources': {'num_machines': 1}}}})
+    assert builder.metadata.options.resources['num_machines'] == 1
+    builder._update({'metadata': {'options': {'resources': {'num_cores_per_machine': 2}}}})
+    assert builder.metadata.options.resources['num_cores_per_machine'] == 2
+    assert 'num_machines' not in builder.metadata.options.resources
+
+
+def test_merge():
+    """Test the ``ProcessBuilder`` merge methods."""
+    builder = ProcessBuilder(ArithmeticAddCalculation)
+
+    builder._update({'metadata': {'options': {'resources': {'num_machines': 1}}}})
+    assert builder.metadata.options.resources['num_machines'] == 1
+    builder._merge({'metadata': {'options': {'resources': {'num_cores_per_machine': 2}}}})
+    assert builder.metadata.options.resources['num_machines'] == 1
+    assert builder.metadata.options.resources['num_cores_per_machine'] == 2
```

Comment:
Are these being added due to a rebase-fu?

Reply:
Nono, I simply wanted to appease the codecov gods here. :)

Reply:
I see. Well I was sure this was already being tested somewhere. It looks like there are two files to test the `ProcessBuilder` and these tests are already in `tests/engine/test_process_builder`.  Think the file you were updating should be the one to keep, so would be good to move the tests from the other there and then you can skip adding these tests as they should be duplicates (largely).

Reply:
This is why the tests folder structure should be the same as the package. 😅

Reply:
So I would suggest to move the `TestProcessBuilder` tests in `tests/engine/test_process_builder` to `tests/engine/process/test_builder` instead.

Reply:
With the risk of being the scope-creeper: the tests of the former are in the old form compared to the pytest style of the latter. Would be might magnificent if you would be willing to update them :innocent:

Reply:
Haha, ok, but only because you _didn't_ ask me to do this in https://github.com/aiidateam/aiida-core/pull/4428/files. 😁

Reply:
![alldone](https://user-images.githubusercontent.com/25607960/151200721-8f0496b6-0112-4438-8f13-204fab88a83a.gif)